enacts/crop_suitability/maproom_crop_suit.py

```diff
@@ -0,0 +1,545 @@
+    data_choice = parse_arg("data_choice")
+    min_wet_days = parse_arg("min_wet_days", int)
+    wet_day_def = parse_arg("wet_day_def", float)
+    lower_wet_threshold = parse_arg("lower_wet_threshold", int)
+    upper_wet_threshold = parse_arg("upper_wet_threshold", int)
+    maximum_temp = parse_arg("maximum_temp", float)
+    minimum_temp = parse_arg("minimum_temp", float)
+    temp_range = parse_arg("temp_range", float) 
+    dry_spell_rain = parse_arg("dry_spell_rain", float)
+    dry_days_in_row = parse_arg("dry_days_in_row",int)
+    number_dry_spells = parse_arg("number_dry_spells", int)    
+
+    x_min = pingrid.tile_left(tx, tz)
+    x_max = pingrid.tile_left(tx + 1, tz)
+
+    # row numbers increase as latitude decreases
+    y_max = pingrid.tile_top_mercator(ty, tz)
+    y_min = pingrid.tile_top_mercator(ty + 1, tz)
+    
+    crop_suit_vals = crop_suitability(rr_mrg,min_wet_days,wet_day_def,tmax_mrg,tmin_mrg,lower_wet_threshold,upper_wet_threshold,maximum_temp,minimum_temp,temp_range,target_season,dry_spell_rain,dry_days_in_row,number_dry_spells) 
+    
+    data_tile = crop_suit_vals.crop_suit
+
+    if (
+            # When we generalize this to other datasets, remember to
+            # account for the possibility that longitudes wrap around,
+            # so a < b doesn't always mean that a is west of b.
+            x_min > data_tile['X'].max() or
+            x_max < data_tile['X'].min() or
+            y_min > data_tile['Y'].max() or
+            y_max < data_tile['Y'].min()
+    ):
+        return pingrid.image_resp(pingrid.empty_tile())
+
+    data_tile = data_tile.sel(
+        X=slice(x_min - x_min % RESOLUTION, x_max + RESOLUTION - x_max % RESOLUTION),
+        Y=slice(y_min - y_min % RESOLUTION, y_max + RESOLUTION - y_max % RESOLUTION),
+    ).compute()
+    mymap_min = float(0) 
+    mymap_max = float(1)
+
+    mycolormap = pingrid.RAINBOW_COLORMAP
+
+    #mymap = data_tile.mean("year")
+    mymap = data_tile[data_tile["year"] == target_year]
+    mymap = np.squeeze(mymap)
+    mymap.attrs["colormap"] = mycolormap
+    mymap = mymap.rename(X="lon", Y="lat")
+    mymap.attrs["scale_min"] = mymap_min
+    mymap.attrs["scale_max"] = mymap_max
+    result = pingrid.tile(mymap, tx, ty, tz, clip_shape)
+
+    return result
+
+
+@APP.callback(
+    Output("colorbar", "children"),
+    Output("colorbar", "colorscale"),
+    Output("colorbar", "max"),
+    Output("colorbar", "tickValues"),
+    Input("data_choice", "value"),
+)
+def set_colorbar(
+    data_choice,
+):
+    mymap_max = 1
+    return (
+        f"{CONFIG['map_text'][data_choice]['menu_label']} [{CONFIG['map_text'][data_choice]['units']}]",
+        pingrid.to_dash_colorscale(pingrid.RAINBOW_COLORMAP),
+        mymap_max,
+        [0,0.5,1],
+    )
+
+if __name__ == "__main__":
+    APP.run_server(
+        debug=GLOBAL_CONFIG["mode"] != "prod",
+        processes=GLOBAL_CONFIG["dev_processes"],
+        threaded=False,
```

Comment:
To me those all look different aside from which options are defined. What you see here I copied from the [onset maproom](https://github.com/iridl/python-maprooms/blob/947d5d075782ab5f76ff15eec85273b3445250f5/enacts/onset/maproom.py#L951)

Then in ../app.py it is:

    if __name__ == "__main__":
        if GLOBAL_CONFIG["mode"] != "prod":
            import warnings
            warnings.simplefilter("error")
            debug = True
        else:
            debug = False

        FLASK.run(
            GLOBAL_CONFIG["dev_server_interface"],
            GLOBAL_CONFIG["dev_server_port"],
            debug=debug,
            extra_files=os.environ["CONFIG"].split(":"),
            processes=GLOBAL_CONFIG["dev_processes"],
            threaded=False,
        )

But in ../wat_bal/maproom_monit.py they are:

    if __name__ == "__main__":
        APP.run_server(
            host=CONFIG_GLOBAL["dev_server_interface"],
            port=CONFIG_GLOBAL["dev_server_port"],
            debug=CONFIG_GLOBAL["mode"] != "prod",
            processes=CONFIG_GLOBAL["dev_processes"],
            threaded=False,
        )

So is it only important to you which config options are defined? Or do you also want me to follow how the rest of this code is written, for example how ../app.py has a warning message while the others do not?

Reply:
@drewmresnick please address this. It's making it harder for me to help you with the branch, because I can't run it on shortfin01 if anyone else is using port 8050.